### ISSUE TITLE
Colors don't react on Day/Night theme change.

### ISSUE THREAD

**Issue by Makzimalist** [leinardi/FloatingActionButtonSpeedDial#152]
### Step 1: Are you in the right place?
- [x] I have verified there are no duplicate active or recent bugs, questions, or requests
- [x] I have verified that I am using the latest version of the library.

### Step 2: Describe your environment
 - Library version: `3.1.1`
 - Android version: `8-10`
 - Support library version: `androidx 1.1.0`

#### Steps to reproduce:

  1.  Implement the SpeedDial.
  2. Set generic colors for MainFab and action items. like : `@color/colorAccent`
  3. change day or night Theme via `AppCompatActivity.delegate.applyDayNight()`
  4. observe that colors doesn't change. App restart is required!

#### Observed Results:

  * Appstart is needed for colors to react on Day/Night switch
  

#### Expected Results:

  * Colors behave like intended and change immediately 

If You need further Infos, please shoot!
Nice project! would be even better if you would switch to kotlin 😜

**Comment by Khareesh0892** [comment]
Same issue.
When a day/night theme is applied i.e `onCreate()` is re-triggered in my case but `mExpandFabView` somehow retains the old theme. But when i close app and restart then it works correctly.

Here is my sample code
```
mExpandFabView.addActionItem(
    new SpeedDialActionItem.Builder(R.id.some_label, R.drawable.ic_text_24dp)
        .setFabImageTintColor(ThemeUtils.getColor(this, R.attr.colorPrimary))
        .setLabelColor(ThemeUtils.getColor(this, R.attr.colorOnPrimary))
        .setLabel("Some Text")
        .setLabelBackgroundColor(ThemeUtils.getColor(this, R.attr.colorPrimary))
        .setTheme(R.style.AppTheme_ExpandFAB)
        .create()
);
```

Code from ThemeUtils
```
public static @ColorInt int getColor(Context context, @AttrRes int i) {
    TypedArray obtainStyledAttributes = context.getTheme().obtainStyledAttributes(new int[]{i});
    int color = obtainStyledAttributes.getColor(0, 0);
    obtainStyledAttributes.recycle();
    return color;
}
```

@leinardi Am i doing something wrong or is that expected? Please help me i am stuck.

**Comment by leinardi** [comment]
Hi, sorry for the late reply but I'm struggling to find some free time lately.

So, I tried to reproduce the issue but I was not able: DayNight theme works fine for me:

![image](https://user-images.githubusercontent.com/273338/100517261-aa289900-3189-11eb-9bb9-99593edbc53f.png)
![image](https://user-images.githubusercontent.com/273338/100517264-af85e380-3189-11eb-908b-adb400b01ded.png)

I just pushed an update of the Sample app that adds a toggle for the DayNight theme. Could you please try to reproduce the issue there?

Also, be sure to use the latest stable Material Component library version.

**Comment by MackHartley** [comment]
Hey there, I just wanted to confirm I am also having this issue. If I set the speed dial color to ?attr/speedDialColor then the color will not change when I switch to dark mode.

I'm currently trying to figure out a way around this using themes, but I can't figure out how to change the mini fab colors via theme. However I'm pretty sure this bug is valid.

**Comment by MackHartley** [comment]
Just an update, I figured out a way around this bug. Just set a theme on your speed dial view and then create a theme like this:
```
<style name="AppTheme.MySpeedDialView" parent="AppTheme">
        <item name="colorAccent">@color/fab_color</item>
        <item name="colorPrimary">@color/fab_color</item>
    </style>
```
It seems like `colorAccent` controls the main fab color and `colorPrimary` controls the mini fab color.

Also thanks a bunch for this library! It's amazing and looks great!

**Comment by MackHartley** [comment]
Hey folks, I've ran into this issue again but for the fab icon color. So far I can't find a way to update the fab icon color when the day / night theme changes. If anyone know's a fix I'd love to hear it.

Edit: I browsed through some old issues and found issue #111  in which someone stated that `colorOnSecondary` changes the fab icon color. That did the trick for me. I simply added this line to the above style:
`<item name="colorOnSecondary">@color/fab_icon_color</item>`
